.instrumentation_generator/test/instrumentation_generator_test.rb

```diff
@@ -0,0 +1,84 @@
+# test/instrumentation_generator_test.rb
+
+require 'minitest/autorun'
+require 'mocha/minitest'
+require 'thor'
+require_relative '../instrumentation_generator'
+
+class InstrumentationGeneratorTest < Minitest::Test
+  def setup
+    @instrumentation_name = 'new_instrumentation'
+    @generator = InstrumentationGenerator.new([@instrumentation_name])
+
+    @generator.stubs(:template)
+    @generator.stubs(:insert_into_file)
+    @generator.stubs(:puts)
+  end
+
+  def test_root_files
+    @generator.root_files
+
+    assert @generator.template('templates/rubocop.yml.tt', 'instrumentation/new_instrumentation/.rubocop.yml')
```

Comment:
First off, thank you for adding tests! It's great to have coverage on a tool that we use whenever we add new instrumentation. 

I like this example as the three options a Ruby unit test should evaluate[^1]: 
* It returns a value.
* It passes the work along to somewhere else (i.e., dispatches the work elsewhere).
* It causes a side effect.

In this test, it's almost as if we're testing whether the `@generator.template` method can be called. Since we don't define `@generator.template`, ([that's from `Thor`](https://www.rubydoc.info/gems/thor/Thor%2FActions:template)), I think we could take a different approach to make sure everything is working as expected.

Of the three options listed above, I think we want to make sure this method causes a side effect.

Since the `root_files` method is supposed to create these files from the ERB templates, what if we checked to see if the files were created when the method is run? 

That could look like: 
`assert File.exist?('instrumentation/new_instrumentation/.yardopts')`

Let me know if you want help brainstorming assertions related to side effects of the other methods.

[^1]: https://www.cloudbees.com/blog/unit-testing-in-ruby#what-should-be-tested